src/ui/js/TournamentOverview.js

```diff
@@ -0,0 +1,366 @@
+// * TournamentOverview.getTournamentOverview() asks the API for information
+//   about the player's tournament overview status (currently, the lists of new,
+//   active, completed, and cancelled tournaments, and potentially any other
+//   tournaments that have been tagged to be watched).
+//   It sets Api.new_tournaments, Api.active_tournaments,
+//   Api.completed_tournaments, Api.cancelled_tournaments, and potentially
+//   Api.tagged_tournaments. If successful, it calls
+//   TournamentOverview.showPage().
+// * TournamentOverview.showPage() assembles the page contents as a variable.
+//
+// N.B. There is no form submission on this page (aside from the [Dismiss]
+// links); it's just a landing page with links to other pages. So it's
+// logically somewhat simpler than e.g. Game.js.
+////////////////////////////////////////////////////////////////////////
+
+TournamentOverview.showLoggedInPage = function() {
+  // Get all needed information, then display overview page
+  TournamentOverview.getOverview(TournamentOverview.showPage);
+};
+
+TournamentOverview.getOverview = function(callback) {
+  Env.callAsyncInParallel([
+    Api.getTournamentsData
+  ], callback);
+};
+
+TournamentOverview.showPage = function() {
+  TournamentOverview.page = $('<div>');
+
+  TournamentOverview.pageAddNewtournamentLink();
+  TournamentOverview.pageAddTournamentTables();
+
+  // Actually lay out the page
+  Login.arrangePage(TournamentOverview.page);
+
+  TournamentOverview.updateSectionHeaderCounts('closedtournaments');
+  TournamentOverview.updateSectionHeaderCounts('opentournaments');
+  TournamentOverview.updateSectionHeaderCounts('activetournaments');
+
+  TournamentOverview.updateVisibilityOfTables();
+};
+
+
+//////////////////////////////////////////////////////////////////////////
+//// Helper routines to add HTML entities to existing pages
+//
+TournamentOverview.pageAddNewtournamentLink = function() {
+  var newtournamentDiv = $('<div>');
+  var newtournamentPar = $('<p>');
+  var newtournamentLink = $('<a>', {
+    'href': Env.ui_root + 'create_tournament.html',
+    'text': 'Create a new tournament',
+  });
+
+  newtournamentPar.append(newtournamentLink);
+  newtournamentDiv.append(newtournamentPar);
+  TournamentOverview.page.append(newtournamentDiv);
+};
+
+// Add tables for types of existing tournaments
+TournamentOverview.pageAddTournamentTables = function() {
+  TournamentOverview.pageAddTournamentTable('closed', 'Closed tournaments');
+  TournamentOverview.pageAddTournamentTable('open', 'Open tournaments');
+  TournamentOverview.pageAddTournamentTable('active', 'Active tournaments');
+};
+
+TournamentOverview.pageAddTournamentTable = function(
+    tournamentType,
+    sectionHeader,
+    reverseSortOrder
+  ) {
+  var tournamentsource = [];
+  var tournamentIdx;
+  var tableClass;
+  var showDismiss = false;
+  var showNPlayersJoined = false;
+
+  switch (tournamentType) {
+  case 'open':
+    // open games that you haven't joined
```

Comment:
Typo: this comment should say "open tournaments"

[I definitely don't care enough to track this for fixing, but fix it if you happen to remember.